app/components/shared/Duration.js

```diff
@@ -23,67 +46,99 @@ class Duration extends React.PureComponent {
   };
 
   static defaultProps = {
-    updateFrequency: 1::second,
+    updateFrequency: second.bind(1),
     tabularNumerals: true
   };
 
-  state = {
-    value: ''
-  };
+  _timeout: TimeoutID;
 
-  updateTime() {
-    const { from, to, format } = this.props;
+  state = {};
+
+  tick() {
+    const { from, to } = this.props;
 
     this.setState({
-      value: getDurationString(from, to, format)
+      seconds: getDuration(from, to).seconds()
+    }, () => {
+      this.maybeScheduleTick();
     });
   }
 
   componentDidMount() {
-    this.maybeSetInterval(this.props.updateFrequency);
+    this.maybeScheduleTick();
   }
 
-  maybeSetInterval(updateFrequency) {
-    if (updateFrequency > 0) {
-      this._interval = setInterval(() => this.updateTime(), updateFrequency);
+  maybeScheduleTick() {
+    const { from, to, updateFrequency } = this.props;
+
+    // We only want to schedule ticks if our duration is indeterminate,
+    // and our update frequency isn't zero
+    if (!(from && to) && typeof updateFrequency == 'number' && updateFrequency > 0) {
+      this._timeout = setTimeout(() => this.tick(), updateFrequency);
```

Comment:
Is the churn of creating new timeouts every time negligible, compared to setting up and tearing down an interval?

Reply:
Timeout setups are super negligible. This is actually just so that we end up with less work queued overall; an interval _can_ end up queueing up work if it takes too long, while this approach means new work is only ever scheduled after the last unit of work has been done (read: React has rendered it)

Reply:
:+1: